source/rock/middle/TypeDecl.ooc

```diff
@@ -1391,6 +1420,9 @@ TypeDecl: abstract class extends Declaration {
                             // if the variable is static, use class scope not instance
                             name := vDecl isStatic() ? "This" : "this"
                             call setExpr(VariableAccess new(name, call token))
+                            if (res params explicitThis) {
+                                call token printMessage("Implicit " + name + " detected for call to " + vDecl name + "!")
```

Comment:
I guess that this is function calls by pointer since a variable is used for access.